org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/parser/Scanner.java

```diff
@@ -4861,12 +4860,12 @@ void updateScanContext(int token) {
 
 	@Override
 	public boolean isParsingModuleDeclaration() {
-		return true;
+		return Scanner.this.insideModuleInfo;
```

Comment:
Note that the only test where this made a difference is org.eclipse.jdt.core.tests.compiler.regression.SwitchExpressionsYieldTest.testBug577220_001() (illegal java code in a module-info.java)

Reply:
Comments provided on the PR

Reply:
So what difference did it make ?? In Jenkins the tests seem to be passing and I don't see any changes to the tests themselves ??

This change and the next in `org.eclipse.jdt.internal.compiler.parser.Scanner.ScanContextDetector.getScanContext(char[], int)` are confusing to me. Why would you replace true with some state in scanner. We get into the method `org.eclipse.jdt.internal.compiler.parser.Scanner.ScanContextDetector.getScanContext(char[], int)` only if we are in module-info. If you wanted to switch the call from `org.eclipse.jdt.internal.compiler.parser.Scanner.resetTo(int, int)` to `org.eclipse.jdt.internal.compiler.parser.Scanner.resetTo(int, int, boolean)` why won't you simply pass true.

Reply:
When debugging the test it was visible, that during isInModuleDeclaration() this.insideModuleInfo was still false even though it was within a module. so the point is that resetTo(0, begin) did not setup this.insideModuleInfo correctly. If this code is called only in a module-info it is equivalent to use `true `or Scanner.this.insideModuleInfo. But its not obvious that that the code should only be used that way.

Reply:
> When debugging the test it was visible, that during isInModuleDeclaration() this.insideModuleInfo was still false even though it was within a module. so the point is that resetTo(0, begin) did not setup this.insideModuleInfo correctly. If this code is called only in a module-info it is equivalent to use `true `or Scanner.this.insideModuleInfo. But its not obvious that that the code should only be used that way.

Please take a look at the first line of `org.eclipse.jdt.internal.compiler.parser.Scanner.getScanContext(int)` which reads:

```
if (!isInModuleDeclaration())
		return ScanContext.INACTIVE;
```

The only instantiation of `org.eclipse.jdt.internal.compiler.parser.Scanner.ScanContextDetector` happens below this. So it is guaranteed that it is created only for module-info files. In fact its very purpose is to handle module-info files. So this change is obscuring matters. Leave it as `return true;` please